### ISSUE TITLE
Remove extra spaces between elements

### ISSUE THREAD

**Issue by evgygor** [TwitchBronBron/brightscript-formatter#20]
Please add formatting for spaces between elements, actually remove extra spaces.

Example.
**The following code**
```
if a >  b  then
   print     "A is   bigger"
end if
```
**should be formatted to**
```
if a > b then
   print "A is   bigger"
end if
```

No spaces removal needed inside the strings.

Would be nice to have same formatting as in Typescript:
'a>b' formatted  to 'a > b'
'foo ()' formatted to 'foo()'

Use same spacing rules from typescript.
But save the following indentation too

```
            self.states[States().CREATED] =     CreatedState()
            self.states[States().INITIATING] =  InitiatingState()
            self.states[States().LOADED] =      LoadedState()
            self.states[States().STARTING] =    StartingState()
            self.states[States().STARTED] =     StartedState()
            self.states[States().PLAYING] =     PlayingState()
            self.states[States().RESUMING] =    ResumingState()
            self.states[States().RESUMED] =     ResumedState()
            self.states[States().PAUSING] =     PausingState()
            self.states[States().PAUSED] =      PausedState()
            self.states[States().STOPPING] =    StoppingState()
            self.states[States().STOPPED] =     StoppedState()
            self.states[States().ERROR] =       ErrorState()
            self.states[States().TERMINATED] =  TerminatedState()
```

Thanks

**Comment by TwitchBronBron** [comment]
So, yes, I definitely want to support your request. It just may take a while. A little background information: this project does not currently use a parser. It basically just scans for a few known keywords, and does some very basic formatting. Without an _actual_ parser, formatting most things gets pretty complicated. 

There is a great brightscript parser included in the [brs](https://github.com/sjbarag/brs) that this project will most likely use in the future, but that would require almost an entire rewrite of brightscript-formatter. 

In the mean time, there are a few other things we could do hold us over. For example, we could find all space tokens that are not contained within strings, and replace them all with single space. i.e. replace SPACESPACESPACE with SPACE. That would fix your first if statement condition. 

However, that would not satisfy your `self.states` requirement. What programming languages have you worked on that supported formatting, yet preserved the spacing like in your `self.states` example? Typescript, for example, will kill your perfect spacing in your example.

**Comment by evgygor** [comment]
> So, yes, I definitely want to support your request. It just may take a while. A little background information: this project does not currently use a parser. It basically just scans for a few known keywords, and does some very basic formatting. Without an _actual_ parser, formatting most things gets pretty complicated.
> 
> There is a great brightscript parser included in the [brs project](https://github.com/sjbarag/brs) that this project will most likely use in the future, but that would require almost an entire rewrite of brightscript-formatter.
> 
> In the mean time, there are a few other things we could do hold us over. For example, we could find all space tokens that are not contained within strings, and replace them all with single space. i.e. replace SPACESPACESPACE with SPACE. That would fix your first if statement condition.
> 
> However, that would not satisfy your `self.states` requirement. What programming languages have you worked on that supported formatting, yet preserved the spacing like in your `self.states` example? Typescript, for example, will kill your perfect spacing in your example.

Support this kind of the alignment related to the IDE, no a language. 
For example, Eclipse doesn't save the alignment like this for Java, but IntelliJ does.

**Comment by TwitchBronBron** [comment]
> Support this kind of the alignment related to the IDE

This project (`brightscript-formatter`) is a standalone tool written in nodejs. It has no access to IDEs at all. 

>removing extra spaces from the code more important. 

I'll try to carve out some time to put the "remove duplicate spaces" feature in.

**Comment by TwitchBronBron** [comment]
This was mostly implemented in release v1.6.0
